File: Platform/codesignal/Intro/20.arrayMaximalAdjacentDifference/solution.go
package main

import (
	"fmt"
)

func solution(arr []int) int {
	result, diff := 0, 0
	for i := 1; i < len(arr); i++ {
		diff = (arr[i] - arr[i-1])
		if diff < 0 {
			diff *= -1
		}
		if diff > result {
			result = diff
		}
	}
	return result
}

func main() {
	arr := []int{2, 4, 1, 0}
	fmt.Print(solution(arr))
}

File: Platform/codesignal/Intro/20.arrayMaximalAdjacentDifference/solution.js
function solution(arr){
  let result = 0
  for(let i=1; i<arr.length; i++) {
    result = Math.max(result, Math.abs(arr[i] - arr[i-1]))
  }
  return result
}

function solution(arr) {
  return Math.max(...arr.slice(1).map((x,i)=>Math.abs(x-arr[i])))
}


console.log(solution([2, 4, 1, 0]));